packages/cli/src/render-solidity/worldgen.ts

```diff
@@ -0,0 +1,57 @@
+import { readFileSync } from "fs";
+import path from "path";
+import { MUDConfig } from "../index.js";
+import { contractToInterface } from "../utils/contractToInterface.js";
+import { formatAndWrite } from "../utils/formatAndWrite.js";
+import { renderSystemInterface } from "./renderSystemInterface.js";
+import { renderWorld } from "./renderWorld.js";
+import { ImportDatum } from "./types.js";
+
+export async function worldgen(
+  config: MUDConfig,
+  existingContracts: { path: string; basename: string }[],
+  outputBaseDirectory: string
+) {
+  const worldgenBaseDirectory = path.join(outputBaseDirectory, config.worldgenDirectory);
+  const systems = existingContracts.filter(({ basename }) => Object.keys(config.systems).includes(basename));
+
+  const systemInterfaceImports: ImportDatum[] = [];
+  for (const system of systems) {
+    const data = readFileSync(system.path, "utf8");
+    // get external funcions from a contract
+    const { functions, symbols } = contractToInterface(data, system.basename);
+    const imports = symbols.map((symbol) => ({
+      symbol,
+      path: "./" + path.relative(worldgenBaseDirectory, system.path),
+    }));
+    const systemInterfaceName = `I${system.basename}`;
+    // create an interface using the external functions and imports
+    const output = renderSystemInterface({
+      name: systemInterfaceName,
+      functionPrefix: config.namespace === "" ? "" : `${config.namespace}_${system.basename}_`,
+      functions,
+      imports,
+    });
+    // write to file
+    const fullOutputPath = path.join(worldgenBaseDirectory, systemInterfaceName + ".sol");
+    await formatAndWrite(output, fullOutputPath, "Generated system interface");
+
+    // prepare imports for IWorld
+    systemInterfaceImports.push({
+      symbol: systemInterfaceName,
+      path: `./${systemInterfaceName}.sol`,
+    });
+  }
+
+  // render IWorld
+  const worldInterfaceName = "IWorld";
```

Comment:
Should we make this configurable? Or make it `I${worldContractName}` and fall back to `IWorld` if `worldContractName` is undefined?

Reply:
I think that would be kind of a lie then, since the custom world might have its own functions, which this interface won't have. Imo this should be done together with support for extra interfaces as its own PR

Reply:
sounds good!